### ISSUE TITLE
ACS FWTS SMBIOS failures

### ISSUE THREAD

**Issue by samerhaj** [pftf/RPi4#75]
There are multiple SMBIOS FWTS/ACS failures in Rel1.15:

https://github.com/pftf/acs-reports/blob/master/v1.15/acs-enterprise-v2.4/luv-results-2020-06-12--16-30-39/parsed/fwts

```
1.1  fwts FAILED INFO [HIGH] DMIStringIndexOutOfRange: Test 3, Out of range string index 0x05
1.1  fwts while INFO accessing entry 'Base Board Information (Type 2)' @ 0x371d00d2, field
1.1  fwts 'Asset INFO Tag', offset 0x08
```
- [x]  Failure when AssetTag is NULL, as that is invalid.

```
1.1  fwts FAILED INFO [HIGH] DMIInvalidHardwareEntry: Test 3, Incorrect Chassis Type SMBIOS
1.1  fwts Type INFO 3 reports 0x22
1.1  fwts ADVICE: INFO It may be worth checking against section 7.4 of the System Management
1.1  fwts BIOS INFO (SMBIOS) Reference Specification (see http://www.dmtf.org/standards
1.1  fwts /smbios). INFO /smbios).
```
- [x] The value 0x22 is Embedded PC, defined in SMBIOS spec 3.3.0. The test is out-of-date. Upstream FWTS has this type defined: https://git.launchpad.net/fwts/tree/src/lib/include/fwts_smbios.h#n66

```
1.1  fwts FAILED INFO [MEDIUM] DMIValueOutOfRange: Test 3, Invalid value 0x0000 was used and
1.1  fwts 0xffffshould INFO be used for non-PCI(e) while accessing entry 'System Slot
1.1  fwts Information INFO (Type 9)' @ 0x371d01e4, field 'Segment Group Number', offset 0x0d
.21

1.1  fwts FAILED INFO [MEDIUM] DMIValueOutOfRange: Test 3, Invalid value 0x00 was used and
1.1  fwts 0xffshould INFO be used for non-PCI(e) while accessing entry 'System Slot Information
1.1  fwts (Type INFO 9)' @ 0x371d01e4, field 'Bus Number', offset 0x0f

1.1  fwts FAILED INFO [MEDIUM] DMIValueOutOfRange: Test 3, Invalid value 0x00 was used and
1.1  fwts 0xffshould INFO be used for non-PCI(e) while accessing entry 'System Slot Information
1.1  fwts (Type INFO 9)' @ 0x371d01e4, field 'Device/Function Number', offset 0x10
```
- [x] These are SMBIOS Type 9 values that should be 0xFFs for non-PCIe devices. Fixing.

The remaining items are SKIPPED tests ( not failures) because of  missing recommended tables
```
1.1  fwts SKIPPED: INFO Test 4, SMBIOS structure Port Connector Information (Type 8) not found.
1.1  fwts Recommended INFO for platforms with physical ports.
```
- [x] Plan on adding Type 8. Not required

```
1.1  fwts SKIPPED: INFO Test 4, SMBIOS structure OEM Strings (Type 11) not found. This
1.1  fwts structure INFO is recommended.
```
- [x] Plan on adding Type11. Not required

```
1.1  fwts SKIPPED: INFO Test 4, SMBIOS structure BIOS Language Information (Type 13) not found.
1.1  fwts This INFO structure is recommended.
```
- [x] Plan on adding Type 13. Not required

```
1.1  fwts SKIPPED: INFO Test 4, SMBIOS structure System Event Log (Type 15) not found. This
1.1  fwts structure INFO is recommended.
```
- [x] No plan to add Type 15. Not required

```
1.1  fwts SKIPPED: INFO Test 4, SMBIOS structure IPMI Device Information (Type 38) not found.
1.1  fwts Required INFO for platforms with IPMI BMC Interface.
```
- [x] No plan to add Type 38. Not required for a system without a BMC. Ignore

```
1.1  fwts SKIPPED: INFO Test 4, SMBIOS structure Onboard Devices Extended Information (Type 41)
1.1  fwts not INFO found. This structure is recommended.
```
- [x] Plan on adding Type 41. Not required

**Comment by samerhaj** [comment]
As of release v1.19, all issues fixed

**Comment by samerhaj** [comment]
Confirmed all fixed in v1.19 FW and ACS 2.5 beta:

https://github.com/pftf/acs-reports/blob/master/v1.19/acs-enterprise-v2.5-beta/luv-results-2020-08-24--10-11-06/raw/fwts#L392

```
dmicheck: DMI/SMBIOS table tests.
...
================================================================================
30 passed, 0 failed, 0 warning, 0 aborted, 6 skipped, 0 info only.
================================================================================
```
